nixos/default.nix

```diff
@@ -1,4 +1,13 @@
+let
+  looks-like-nixos-lib = { path, prefix }:
+    prefix == "nixos-lib" ||
+    (prefix == "" && builtins.pathExists "${path}/nixos-lib");
+in
+
 { configuration ? import ./lib/from-env.nix "NIXOS_CONFIG" <nixos-config>
+, lib ? if builtins.any looks-like-nixos-lib builtins.nixPath
+          then import <nixos-lib>
+          else import ../lib
```

Comment:
Oh actually, we can just use `lib ? import <nixpkgs/lib>` here. Then users can override it by setting

```
NIX_PATH=nixpkgs/lib=/path/to/lib:nixpkgs=/path/to/nixpkgs
```

This is then a single-line change pretty much accomplishing the same :)

Reply:
Rather than setting lib like this, I think they would prefer to use the modular way (as mentioned in various comments), setting `_module.args.lib`. This requires some changes to `lib/modules.nix` to not make lib a special case any more, but I think this would be more in line with what we already have for e.g. pkgs.

Reply:
I wasn't aware `nixpkgs/lib` can be overridden :heart: